### ISSUE TITLE
Theia is RIDICULOUSLY slow and crashes often

### ISSUE THREAD

**Issue by sarthakpati** [CBICA/CaPTk#129]
Hopefully, when this gets moved to a C++ implementation, it will be better.

**Comment by PhucNgo1711** [comment]
Yes these are existing issues. Probably because it's in Python or the way the code is implemented. I'll take a look.
